### PR TITLE
chore: update redactor

### PR DESCRIPTION
**Pull request by ningsongshen**
The original (working) Redactor was on v3.1.8. This version is over a year old and should be updated.

Edit: refer to the later comments. This comment is outdated.

> <del>Yesterday, I updated the Redactor files to the latest version (3.4.1, released ~1wk ago), and there are some new (and different) issues with images. I cannot paste new images (as `imageUpload` isn't getting called) and "Copy Insert Link" only provides Filesafe tags: `[FileSafe:4a1abed6-ec21-4151-a71e-badb6448b0f2:image.png]` which are loaded when I switch away and come back to the note.~~
> 
> ~~However, _the last Bold Editor release did not use v3.4.1_ so Redactor was not the issue. In the last release, v3.3.5 was used and the issue was something different: images do not persist after pasting and refreshing.~~
> 
> ~~But v3.4.1 introduces some new issues (not sure if its a Redactor bug or Bold Editor bug) so future updates need to be aware of this. Here are the potentially relevant changelogs from https://imperavi.com/redactor/log/~~
> 

> - Redactor 3.4.0
>     - Improved: Plenty of improvements in paste content.
>     - Improved: Many improvements to protect against XSS.
> - Redactor 3.4.1~~
>  - Fixed: In some cases, drag and drop images didn't work.
> - The `imageUpload` setting is not being called when we try to paste an image, rather, the `pasted` callback is (which is for text).~~

</del>

### CODE REVIEW

**Comment by ningsongshen** [comment]
Good news, they fixed the 'image pasting from clipboard' issues: https://imperavi.com/redactor/log/

So now it is only the 'image persistence after refreshing' issue that is left.

**Comment by ningsongshen** [comment]
This change seems to make the image uploads particularly robust...and I don't see any unintended side effects&mdash;yet :relieved:  

@mobitar would you happen to recall the purpose of that section? The comments don't explain why it's necessary.

**Comment by ningsongshen** [comment]
I did some more testing today and believe that the issues with Redactor have been resolved. Image uploads, image persistence, image resizing work like in the previous version.

File: app/components/Editor.js [hunk 1]
```diff
@@ -34,15 +34,7 @@ export default class Editor extends React.Component {
         for(let attribute of element.attributes) {
           newElement.setAttribute(attribute.nodeName, attribute.nodeValue);
         }
-
-        if(newElement.tagName != element.tagName) {
```
Comment (mobitar) [inline app/components/Editor.js:38]: @ningsongshen  Can you explain your reasoning as to why you think we don't need this anymore? I forget exactly even how to test this usecase, so I'm curious how you tested it..

Comment (mobitar) [reply app/components/Editor.js:38]: Oh sorry just noticed you mentioned this in your comments. Let me investigate..

Comment (mobitar) [reply app/components/Editor.js:38]: Hmm, it looks like resizing an image no longer persists its updated size. Even if you make another change to trigger a save, the image always retains its original size.

**Comment by mobitar** [comment]
Everything seems to be working great 👍 Only two issues I noticed are:

- When drag and dropping image, it always attaches to top of note, regardless of where it was dropped. This is a pre-existing issue I believe and not a consequence of this PR.
- Image resizing works functionally, but the size of the image is not persisted when you come back.

File: app/components/Editor.js (hunk 1 above)
Comment (ningsongshen) [reply app/components/Editor.js:38]: It seems like we don't need it because all the information is contained in the `` tag. From what I see, the `<figure>` tag is mainly to help Redactor identify it for image resizing and positioning purposes.

**Comment by ningsongshen** [comment]
I removed the loop that copies over attributes and the `width` attribute is now being kept :+1: . It must have something to do with Redactor cleaning. We shouldn't need the loop because `img` contains the necessary info. Also checked if it was the preProcessElement function is used elsewhere---answer is yes, but those are not affected (text based).

Here are the logs showing the difference:
[testBefore.txt](https://github.com/standardnotes/bold-editor/files/4968222/testBefore.txt)
[testAfter.txt](https://github.com/standardnotes/bold-editor/files/4968223/testAfter.txt)

In testAfter, the style/width attribute is maintained while the other figure attributes are not maintained. I'm still unsure why it was necessary in the first place, but things seem to work fine without it. The figure tag just helps redactor with captions, resizing, etc. (see above comment).

So in summary, image size is now persisted!

**Comment by mobitar** [comment]
Seeing issues with this:

1. Insert image via copy insert link and paste
2. Switch to non-Bold editor note
3. Return to Bold editor note

Image now tries to load from `blob:http://localhost:8080/50650f2b-e0cf-476f-b369-571c40c33373` and fails.

> I'm still unsure why it was necessary in the first place

Always assume that every piece of code is probably necessary ;) Unfortunate that I didn't add more comments to clarify why and now the reason is probably lost forever (until we replicate the issue where we discover why it was needed).

**Comment by ningsongshen** [comment]
**The Fix**
Ok I moved the code back to when 'everything seems to be working great' :tada: and then specifically addressed the image resizing. 

By copying the `height` and `width` attributes from the `img` to the `figure` in the `imageResized` callback, the sizing is now saved.

**Other Discovery**
I spent a bit of time trying to figure out why the attribute-copying loop was there, but I discovered some other things instead:
- We used to keep all attributes in the `img` tag, which worked and kept height/width/filesafe attributes
- Not sure why we copied over attributes, the images still saved without it (still keeping it anyway!)
- Recent Redactor update improved XSS image cleaning, and now removes all attributes except for standard ones (src, height, width, etc.)
- So it is imperative we copy over attributes to the wrapped `figure` tag, that way FileSafe can still parse and save images
- Img resizing changes attrs on img only, which used to work fine, but now with XSS cleaning, FileSafe *only* parses `figure` tags (`img` tags have no Filesafe)and that's why we need to copy over tags in the latest commit.

File: app/components/Editor.js (hunk 1 above)
Comment (ningsongshen) [reply app/components/Editor.js:38]: Fixed in latest commit

**Comment by mobitar** [comment]
Seems to be all working 👍
